[Cloud] Integrations for telemetry exports

charislam opened the pull request:
# Description

Integrations for telemetry exports to DataDog and AWS CloudWatch.

- [x] Take screenshots

# Links

Fixes https://github.com/timescale/docs-private/issues/32

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Are procedure and highlight tags used appropriately?
- [ ] Has the index been updated appropriately?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] Are all links provided in reference style, and resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

charislam committed:
[Cloud] Integrations for telemetry exports

Integrations for telemetry exports to DataDog and AWS CloudWatch.

charislam commented on cloud/integrations.md:
```diff
@@ -0,0 +1,141 @@
+1.  Enter your AWS credentials. To get your AWS keys, see [Getting access keys
+    for AWS CloudWatch][aws-access-keys].
+1.  Select an AWS Region for your CloudWatch instance.
+1.  **(optional)** Define an IAM role to upload segments to a
+    different account.
```
@lukas-bernert Can you clarify what "upload segments to a different account" means here?

lukas-bernert replied:
If you have two AWS accounts (e.g. one for production and one for development) it makes sense to have users for each developer in the dev account. However, you might not want to give every developer a user for the production account. Instead you can create an IAM role for production with restricted access. Through this role developers can now access the production account.

Better explanation can be found here: https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html

Loquacity commented on cloud/integrations.md:
```diff
@@ -0,0 +1,141 @@
+
+<procedure>
+
+### Creating a data exporter to DataDog
```
This reads weirdly to me. What if we change the titles to "Export|ing data to [tool name]"?

Loquacity commented on cloud/integrations.md:
```diff
@@ -0,0 +1,141 @@
+1.  Enter your AWS credentials. To get your AWS keys, see [Getting access keys
+    for AWS CloudWatch][aws-access-keys].
+1.  Select an AWS Region for your CloudWatch instance.
+1.  **(optional)** Define an IAM role to upload segments to a
```
```suggestion
1.  **OPTIONAL** Define an IAM role to upload segments to a
```

Loquacity commented on cloud/integrations.md:
```diff
@@ -0,0 +1,141 @@
+You can export your service telemetry to a third-party monitoring tool, such as
+[DataDog][datadog] or [AWS CloudWatch][cloudwatch]. To do so, create a data
+exporter and attach your service to it.
+
```
I wonder if we need to document this in the form of process/procedure, instead of just procedures?

charislam committed:
Rearrange from processes to procedures

Rearrange so creating and attaching exporters is documented as processes rather than procedures.

Loquacity commented on cloud/integrations.md:
```diff
@@ -60,8 +72,8 @@ Create a data exporter to send telemetry data to your preferred provider.
 1.  Select an AWS Region for your CloudWatch instance.
-1.  **(optional)** Define an IAM role to upload segments to a
-    different account.
+1.  **OPTIONAL** Define an Identity and Access Management (IAM) role to upload
```
```suggestion
1.  **OPTIONAL** Define an identity and access management (IAM) role to upload
```

Only proper nouns take initcaps 😉

charislam replied:
I think it's the name of an actual thing in AWS and they style it with caps: https://aws.amazon.com/iam/ :)

Loquacity replied:
ugh. OK 🤦‍♀️

charislam committed:
Specify which metrics are exported

lukas-bernert commented on cloud/integrations.md:
```diff
@@ -0,0 +1,136 @@
+
+### Getting access keys for AWS CloudWatch
+
+1.  Navigate to your AWS Console.  
```
there are a couple more steps to this

1. go to AWS console
2. open "IAM" console
3. in navigation pane select "users" --> "add user"
4. give user a user name
5. ... steps you already described

also, this is the link to the AWS docs in case we want to link there instead
https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html#id_users_create_console

charislam committed:
Correct and minor improvements

lukas-bernert commented on cloud/integrations.md:
```diff
@@ -0,0 +1,130 @@
+    docs][cloudwatch-log-naming].
+1.  Enter your AWS credentials. To get your AWS keys, you need to create a new
+    Identity and Access Management (IAM) user in your AWS console. For
+    instructions, see the [AWS documentation][aws-access-keys].
```
can you add a warning saying that they should create a users with restricted access to only Cloudwatch. and that they should treat those keys with care.

charislam committed:
Add warning about securing AWS keys

lukas-bernert commented on cloud/integrations.md:
```diff
@@ -0,0 +1,137 @@
+    your new user has restricted access to only Cloudwatch, and keep your keys
+    secret. For instructions, see the [AWS documentation][aws-access-keys].
+1.  Select an AWS Region for your CloudWatch instance.
+1.  **OPTIONAL** Define an IAM role if you want to upload metrics to another
```
possible to change this to something along the lines of:
"Optional: IAM role to use for uploading metrics. Having a dedicated role with only CloudWatch permissions is a recommended security practice."

charislam committed:
Improve description of IAM role field

charislam committed:
Add info about where to view metrics

Add a few lines about where to go see the metrics in Datadog and
CloudWatch.

charislam committed:
Add variables to includes

Use variables to add the integration name to the partials. This creates
unique IDs for the partial headings so we can link to them from the
process steps.
